jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/MessageInputStream.java

```diff
@@ -155,8 +156,21 @@ public void close()
             if (remainingContent)
                 LOG.warn("MessageInputStream closed without fully consuming content {}", session);
 
-            state = State.CLOSED;
+
+            // Release any buffers taken from the pool.
+            if (activeBuffer != null && activeBuffer != EOF)
```

Comment:
can you add javadoc to the activeBuffer field saying that it is the head of the buffers queue (I think that is right?)